### DISCUSSION TITLE
is it possible to get access to the 'state' of a widget if all i have is its ID?

### DISCUSSION THREAD

**Discussion by pm100** [emilk/egui#4804] [General]
General question, but my specific need is getting info about the cursor location of a TextWidget before I draw it. I hoped to find a context function that would return a state object but did not find it

**Comment by emilk** [comment]
Yes; see https://docs.rs/egui/latest/egui/widgets/text_edit/struct.TextEditState.html#method.load

**Reply by pm100** [reply]
perfect - thanks, also TextEdit.load_state - didnt look hard enough
